backend/src/src-cvesearch/src/main/java/org/eclipse/sw360/cvesearch/datasink/VulnerabilityConnector.java

```diff
@@ -52,14 +53,11 @@ public VulnerabilityConnector() throws IOException{
     // get objects from database     //
     ///////////////////////////////////
     public Optional<Release> getRelease(String releaseId){
-        return Optional.ofNullable(fillVendor(releaseRepository.get(releaseId)));
-    }
-
-    private Release fillVendor(Release release){
-        if (release != null) {
-            vendorRepository.fillVendor(release);
-        }
-        return release;
+        return Optional.ofNullable(releaseRepository.get(releaseId))
+                .map(r -> {
+                    vendorRepository.fillVendor(r);
+                    return r;
```

Comment:
if you make `VendorRepository.fillVendor()` return the filled release, you can make this a one-liner